src/RVRP_data.jl

```diff
@@ -34,43 +28,61 @@ mutable struct Depot
     time_windows::Vector{TimeWindow} # optional
 end
 
-mutable struct Pickup
+mutable struct PickupPoint
     id::String # If its part of a shipment, it has is own id anyway
-    index::Int # Not given in JSON. If its part of a shipment, the index of the shipment
+    index::Int # Not given in JSON. 
     location::Location
-    shipment_id::String # NULL if none
-    price_reward::Float64
-    capacity_request::Float64
     time_windows::Vector{TimeWindow} # optional
-    service_time::Float64 # optional
+    service_time::Float64 # optional; to measure access time for instance
 end
 
-mutable struct Delivery
+
+mutable struct DeliveryPoint
     id::String # If it is part of a shipment,it has is own id anyway
-    index::Int # Not given in SON. if it is part of a shipment, the index of the shipment
+    index::Int # Not given in JSON. 
     location::Location
-    shipment_id::String # NULL if none
-    price_reward::Float64
-    capacity_request::Float64
     time_windows::Vector{TimeWindow} # optional
-    service_time::Float64 # optional
+    service_time::Float64 # optional; to measure access time for instance
+end
+
+mutable struct Product
+    id::String 
+    index::Int # Not given in JSON. 
+    conflicting_product_ids::Vector{String} # if any
+    prohibited_predecessor_product_ids::Vector{String}  # if any
+end
+
+mutable struct Commodity
+    id::String 
+    index::Int # Not given in JSON. 
+    product_id::String
+    sources::Dict{PickupPoint,Float64} # available quantities; undefined if no restrictions, i.e, if available in any PickupPoint in large quantities
+    destinations::Dict{DeliveryPoint,Float64} # capacities; undefined if no restrictions, i.e, if can go to any DeliveryPoint in large quantities
 end
 
-mutable struct Shipment
+mutable struct ShipmentRequest
     id::String
     index::Int # Not given in JSON
-    price_reward::Float64
-    pickup::Pickup
-    delivery::Delivery
-    max_duration::Float64
+    request_type::Int # 1 = pickup, 2 = delivery, 3 = pickup&delivery, 4 = complex request
+    commodity_id::String  
+    is_optional::Bool  # default is false
+    price_reward::Float64 # if is_optional
+    quantity::Float64
+    split_fulfilment::Bool  # default is false
+    precedence_restriction::Int # default is 0 = no restriction; 1 after all pure pickups, 2 after all pure deliveries, 3 if cannot follow a product that is in the prohibited predecessor list
+    pickup_points::Vector{PickupPoint} # defined only if it is a subset of commodity sources; singleton for request_type = pickup or pickup&delivery
+    delivery_points::Vector{DeliveryPoint} # defined only if it is a subset of commodity destinations; singleton for request_type = delivery or pickup&delivery
```

Comment:
Even if there is only one DeliveryPoint (PickupPoint) we still need to define it. Otherwise how do we get the single Point info?

Reply:
other owe copy that of commodity